### PR TITLE
bpo-33649: Fix asyncio-dev

### PR DESCRIPTION
**Pull request by 1st1**
https://bugs.python.org/issue33649

### CODE REVIEW

**Review by asvetlov — APPROVED** [review]

File: Doc/library/asyncio-dev.rst [hunk 1]
```diff
@@ -2,415 +2,237 @@
...
+Debug Mode
+==========
 
-.. versionchanged:: 3.7
+By default asyncio runs in production mode.  In order to ease
+the development asyncio has a *debug mode*.
 
-   The new ``-X dev`` command line option can now also be used to enable
-   the debug mode.
+To enable debugging for an application:
```
Comment (willingc) [inline Doc/library/asyncio-dev.rst:24]: Perhaps subtitles for Enable debug using Environment Variable; Enable debug  - command line option; Enable debug programmatically

File: Doc/library/asyncio-dev.rst [hunk 2]
```diff
@@ -2,415 +2,237 @@
...
+To schedule a callback from a different OS thread, the
 :meth:`loop.call_soon_threadsafe` method should be used. Example::
 
     loop.call_soon_threadsafe(callback, *args)
 
-Most asyncio objects are not thread safe. You should only worry if you access
-objects outside the event loop. For example, to cancel a future, don't call
-directly its :meth:`Future.cancel` method, but::
+Almost all asyncio objects are not thread safe, which is typically
+not a problem unless there is code that works with them from an outside
```
Comment (willingc) [inline Doc/library/asyncio-dev.rst]: ~~an~~

File: Doc/library/asyncio-dev.rst [hunk 3]
```diff
@@ -2,415 +2,237 @@
...
-     result = future.result(timeout)  # Wait for the result with a timeout
-
-The :meth:`loop.run_in_executor` method can be used with a
-:class:`concurrent.futures.ThreadPoolExecutor` to execute a callback in
-different thread so as not to block the event loop's main thread.
+     async def coro_func():
+          return await asyncio.sleep(1, 42)
 
-.. seealso::
+     # Later in some OS thread:
```
Comment (willingc) [inline Doc/library/asyncio-dev.rst]: `# Later in another OS thread`

**Review by willingc — APPROVED** [review]
A few small typos or rewording

File: Doc/library/asyncio-dev.rst (hunk 1 above)
Comment (1st1) [reply Doc/library/asyncio-dev.rst:24]: I'll try to restructure in a follow up PR later! Thanks!

**Pull request merged**